app/src/main/java/com/example/triptracker/view/profile/UserProfileFollowers.kt

```diff
@@ -30,92 +33,102 @@ import androidx.compose.ui.text.style.TextAlign
 import com.example.triptracker.view.Navigation
 import com.example.triptracker.view.NavigationBar
 import com.example.triptracker.view.theme.md_theme_light_dark
 import com.example.triptracker.viewmodel.UserProfileViewModel
+import com.example.triptracker.viewmodel.loggedUser
 
 /** This composable function displays the user's following list. */
 @Composable
 fun UserProfileFollowers(
     navigation: Navigation,
-    userProfileViewModel: UserProfileViewModel = UserProfileViewModel(),
-    userProfile: UserProfile
+    viewModel: UserProfileViewModel = UserProfileViewModel(),
 ) {
-  Scaffold(
-      topBar = {
-        Row(
-            modifier = Modifier.height(100.dp).fillMaxWidth(),
-            verticalAlignment = Alignment.CenterVertically,
-            horizontalArrangement = Arrangement.Center) {
-              Column(modifier = Modifier.padding(5.dp).height(60.dp).width(100.dp)) {
-                // Button to navigate back to the user profile
-                Button(
-                    onClick = { navigation.goBack() },
-                    colors =
-                        ButtonDefaults.buttonColors(
-                            containerColor = Color.Transparent, contentColor = md_theme_light_dark),
-                    modifier = Modifier.testTag("GoBackButton")) {
-                      Icon(imageVector = Icons.Filled.ArrowBack, contentDescription = "Back")
-                    }
-              }
-              Column(
-                  modifier = Modifier.fillMaxWidth().padding(top = 25.dp),
-              ) {
-                Text(
-                    text = "Followers",
-                    style =
-                        TextStyle(
-                            fontSize = 24.sp,
-                            lineHeight = 16.sp,
-                            fontFamily = FontFamily(Font(R.font.montserrat)),
-                            fontWeight = FontWeight(700),
-                            color = Color.Black,
-                            textAlign = TextAlign.Center,
-                            letterSpacing = 0.5.sp,
-                        ),
-                    modifier =
-                        Modifier.width(250.dp)
-                            .height(37.dp)
-                            .padding(5.dp)
-                            .testTag("FollowersTitle"))
-                Box(modifier = Modifier.size(60.dp))
-              }
+  val userMail: String = loggedUser.email ?: ""
+  var userProfile by remember { mutableStateOf(UserProfile("")) }
+  var readyToDisplay by remember { mutableStateOf(false) }
+  var isSearchActive by remember { mutableStateOf(false) }
+
+  // val list = viewModel.userProfileList.value
+  viewModel.getUserProfile(userMail) { profile ->
+    if (profile != null) {
+      userProfile = profile
+      readyToDisplay = true
+    }
+  }
+  when (readyToDisplay) {
+    false -> {
+      // Display a loading screen while the user profile is being fetched
+      Text("Loading...")
```

Comment:
You could use the WaitingScreen function in `view/WaitingScreen.kt` to display an animated loading screen.
Not mandatory tho